### PR TITLE
Use the same registration process as GCM for Firebase

### PR DESCRIPTION
**Pull request by collinjackson**

### CODE REVIEW

**Comment by abarth** [comment]
LGTM

**Comment by collinjackson** [comment]
This has changed a bit to fix the engine build on Mac, can you take a second look @abarth?

/cc @mpcomplete

File: sky/tools/gn [hunk 1]
```diff
@@ -120,6 +123,9 @@ def parse_args(args):
   parser.add_argument('--target-sysroot', type=str)
   parser.add_argument('--toolchain-prefix', type=str)
 
+  parser.add_argument('--enable_firebase', action='store_true', default=False)
+  parser.add_argument('--enable_gcm', action='store_true', default=False)
```
Comment (abarth) [inline sky/tools/gn]: Nit: We seem to be using `-` instead of `_` for these flags.

**Comment by abarth** [comment]
LGTM

**Pull request merged**